### PR TITLE
feat(benchpress): add mean frame time metric

### PR DESCRIPTION
**Pull request by goderbauer**
This is useful to measure the smoothness of animations and scrolling actions.

part of #821

### CODE REVIEW

File: modules/benchpress/src/metric/perflog_metric.ts [hunk 1]
```diff
@@ -184,9 +198,37 @@ export class PerflogMetric extends Metric {
         markStartEvent = event;
       } else if (StringWrapper.equals(ph, 'e') && StringWrapper.equals(name, markName)) {
         markEndEvent = event;
+      } else if (StringWrapper.equals(ph, 'b') &&
```
Comment (tbosch) [inline modules/benchpress/src/metric/perflog_metric.ts]: You should move this into the `if` call further down in Line 221: You don't want to capture start/end marks outside of the `execute` function and don't want to capture events from other processes (which is very unlikely though)...

File: modules/benchpress/src/webdriver/chrome_driver_extension.ts [hunk 2]
```diff
@@ -91,20 +93,42 @@ export class ChromeDriverExtension extends WebDriverExtension {
...
         ListWrapper.push(normalizedEvents, normalizeEvent(event, {'name': name}));
+
       } else if (StringWrapper.equals(cat, 'v8')) {
         if (StringWrapper.equals(name, 'majorGC')) {
           if (StringWrapper.equals(ph, 'B')) {
             majorGCPids[pid] = true;
           }
         }
+
+      } else if (StringWrapper.equals(cat, 'benchmark')) {
```
Comment (tbosch) [inline modules/benchpress/src/webdriver/chrome_driver_extension.ts]: This is a little but too general as a marker. How about `framesBenchmark`, or just `fps` (e.g. used as `console.time('fps')`...

**Comment by tbosch** [comment]
Could you add a top level comment on `ChromeDriverExtension` which `traceCategories` should be set to use this new metric? And ity would be nice to add the already used ones `v8,blink.console,disabled-by-default-devtools.timeline` as well.

File: modules/benchpress/src/webdriver/chrome_driver_extension.ts (hunk 2 above)
Comment (goderbauer) [inline modules/benchpress/src/webdriver/chrome_driver_extension.ts]: `benchmark` is the name Chrome tracing is using as a category for the events I am looking for. I cannot change that as it is set by chrome. Below, I am filtering for the exact event name (`BenchmarkInstrumentation::ImplThreadRenderingStats`) within that category.

**Comment by goderbauer** [comment]
Updated PR after addressing comments. PTAL. Thanks.

**Pull request closed without merging**